aviatrix/resource_aviatrix_smart_group.go

```diff
@@ -15,6 +16,10 @@ var (
 	// dns1123FmtRe is a regular expression that matches dns label names according to rfc 1123.
 	// K8s resource names must adhere to this.
 	dns1123FmtRe = regexp.MustCompile(`^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?$`)
+	// nonTagKeys are the selector key names that don't expect a "tags" key in the selector
+	nonTagKeys = []string{goaviatrix.ExternalKey, goaviatrix.S2CKey, goaviatrix.CidrKey, goaviatrix.FqdnKey}
```

Comment:
```suggestion
	// exclusiveFilterKeys are the selector key names that cannot be used with any other filters in a single
	// match_expressions block.
	exclusiveFilterKeys = []string{goaviatrix.ExternalKey, goaviatrix.S2CKey, goaviatrix.CidrKey, goaviatrix.FqdnKey}
```

You should probably also add `ExternalKey` to the docs https://github.com/AviatrixSystems/terraform-provider-aviatrix/blob/ce4bf8145e0e3d72e3783cf1038e3a8ee3964864/docs/resources/aviatrix_smart_group.md#required